src/Folder.php

```diff
@@ -68,6 +69,16 @@ public static function find_or_make($folderPath)
                 'Name' => array($partSafe, $part)
             ))->first();
 
+            // When in archived mode, find or make should not find folders that don't exist
+            // We check explicitly for Versioned and if it exists then we'll confirm the reading mode isn't archived
+            $versioned = Injector::inst()->get('SilverStripe\Versioned\Versioned');
```

Comment:
```suggestion
            $versioned = Injector::inst()->get(Versioned::class);
```

Reply:
The intent here is to decouple it _slightly_ from versioned, this allows us to skip needing Versioned to exist by referencing the FQN for if it does exist. That addition/change was largely driven by the conversation above with the worry of tight coupling

Reply:
You won't get an error if you reference a class that doesn't exists. So using `Versioned::class` doesn't increase the coupling between asset and versioned.